service/service.go

```diff
@@ -461,6 +472,23 @@ func (r *RunningDriver) Start() (string, error) {
 	return r.listenAddress, nil
 }
 
+// portOnly attempts to return port fragment of address
+func portOnly(address string) (int, error) {
+	portParseErr := fmt.Errorf("failed to parse port from address [%s]", address)
+	colonIndex := strings.IndexByte(address, ':')
+	if colonIndex == -1 {
+		return -1, portParseErr
+	}
+
+	portString := address[colonIndex+1:]
+	port, err := strconv.Atoi(portString)
```

Comment:
We should be safe with the port number since this is a helper, use something like [this](https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/endpoint.go#L117)